vidarr-cli/src/main/java/ca/on/oicr/gsi/vidarr/cli/TestValidatorScript.java

```diff
@@ -19,15 +19,22 @@ public final class TestValidatorScript extends TestValidator {
   private String outputMetrics;
 
   @Override
-  public Validator createValidator() {
+  public Validator createValidator(String outputDirectory, String id) {
     try {
-      final var tempDir = Files.createTempDirectory("vidarr-test-script");
-      final var calculateScript = tempDir.resolve("calculate");
-      final var calculateDir = tempDir.resolve("output");
-      Files.createDirectories(calculateDir);
-      Files.copy(Path.of(metricsCalculate), calculateScript);
+      // If output directory provided, use that else we create temporary directory /tmp
+      final var finalDir = (outputDirectory != null) ? Path.of(outputDirectory)
+          : Files.createTempDirectory("vidarr-test-script");
+      final var finalCalculateScript =
+          (outputDirectory != null) ? Path.of(outputDirectory, id) : finalDir.resolve("calculate");
```

Comment:
You already have a `Path.of(outputDirectory)` called `finalDir` from line 25, no need to keep calling `Path.of(outputDirectory, whatever)` again, use finalDir.resolve like your else case